sdk/python/feast/infra/online_stores/datastore.py

```diff
@@ -32,6 +35,9 @@
 from feast.protos.feast.types.Value_pb2 import Value as ValueProto
 from feast.repo_config import FeastConfigBaseModel, RepoConfig
 from feast.usage import log_exceptions_and_usage, tracing_span
+from feast.utils.generic_utils import AtomicCounter
```

Comment:
We've generally tried to avoid adding `utils` modules since it's a kitchen sink/code smell. If this `AtomicCounter` is only used in this datastore class, maybe we can just define it here? Or rename `feast.utils.generic_utils` to `feast.atomic` or something?

Reply:
I'm actually glad to hear this :) I wanted to use the patterns of the codebase. I'm going to move it into an innerclass just for the `_delete_all_values` since it's only used for that logging statement (or we could remove the logging), and it can always be migrated to increasing scope if needed.